sound/soc/sof/intel/hda-dsp.c

```diff
@@ -297,10 +297,7 @@ static int hda_suspend(struct snd_sof_dev *sdev, int state,
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
 	if (IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC) && runtime_suspend)
```

Comment:
just remove IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC)

Reply:
why? jack detection doesn't really make sense for HDMI, does it?

Reply:
please check hda_codec_jack_wake_enable,  the CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC is there.

#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC)
hda_codec_jack_wake_enable(){
....
}
#else
hda_codec_jack_wake_enable() {}
#endif

Reply:
?? what are you trying to say?

Reply:
@plbossart: @RanderWang is saying, that you don't need to check for CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC before calling hda_codec_jack_wake_enable() because if the CONFIG_... isn't selected, the function is defined as a dummy, so, it's still safe to call it.  @RanderWang you're right, I'll update that, thanks

Reply:
thanks for the translation. I'd like this PR to be completed soonish so that we can move on.

Reply:
thanks @lyakh  for your explanation. Actually, this idea was suggested by @plbossart when I implemented hda_codec_jack_check.